apps/meteor/server/models/raw/Messages.ts

```diff
@@ -1362,22 +1363,28 @@ export class MessagesRaw extends BaseRaw<IMessage> implements IMessagesModel {
 			return count;
 		}
 
-		const messagesToDelete = (
-			await this.find(query, {
-				projection: {
-					_id: 1,
-				},
-				limit,
-			}).toArray()
-		).map(({ _id }) => _id);
+		const messagesToDelete = await this.find(query, {
+			projection: {
+				_id: 1,
+				_hidden: 1,
+				t: 1,
+				editedAt: 1,
+				editedBy: 1,
+			},
+			limit,
```

Comment:
I believe we are forgetting to apply this change when there is no limit defined, right?

Reply:
Oopsss, thanks. great catch :grimacing: 
Fixed!